controllers/pkg/reconcilers/approval/reconciler_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023 The Nephio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package approval
+
+import (
+	"testing"
+	"time"
+
+	porchapi "github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestShouldProcess(t *testing.T) {
+	testCases := map[string]struct {
+		pr             porchapi.PackageRevision
+		expectedPolicy string
+		expectedShould bool
+	}{
+		"draft with no annotation": {
+			pr:             porchapi.PackageRevision{},
+			expectedPolicy: "",
+			expectedShould: false,
+		},
+		"draft with policy annotation": {
+			pr: porchapi.PackageRevision{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"approval.nephio.org/policy": "initial",
```

Comment:
Shouldn't be better to use the constant here?
```suggestion
						PolicyAnnotationName: "initial",
```